securedrop/tests/test_2fa.py

```diff
@@ -18,19 +18,24 @@
 
 @contextmanager
 def totp_window():
-    # To ensure we have enough time during a single TOTP window to do the
-    # whole test, optionally sleep.
+    '''To ensure we have enough time during a single TOTP window to do the
+       whole test, optionally sleep.
+    '''
+
     now = datetime.now()
-    num_secs_elapsed_in_time_window = now.second % 30
-    num_secs_left_in_time_window = 30 - num_secs_elapsed_in_time_window
-
-    if num_secs_left_in_time_window < 5:
-        time.sleep(30 - num_secs_elapsed_in_time_window)
-        now = datetime.now()
-        window_end = now.replace(microsecond=0) + timedelta(seconds=30)
-    else:
-        window_end = now.replace(microsecond=0) + \
-            timedelta(seconds=(30 - num_secs_left_in_time_window))
+    # `or 30` to ensure we don't have a zero-length window
+    seconds_left_in_window = (now.second % 30) or 30
+
+    window_end = now.replace(microsecond=0) + \
+        timedelta(seconds=seconds_left_in_window)
+    window_end_delta = window_end - now
+
+    # if we have less than 5 seconds left in this window, sleep to wait for
+    # the next window
+    if window_end_delta < timedelta(seconds=5):
+        time.sleep(window_end_delta.second +
```

Comment:
looks like `window_end_delta` is a timedelta, and timedelta objects don't have `second` or `microsecond` attributes, so we'll get an `AttributeError`